File: sourcecode/Angular/client/src/app/_models/message.ts
export interface Message{
    id: number;
    senderId: number;
    senderName: string ;
    senderAvatarPath: string ;
    recipientId: number;
    recipientName:  string ;
    recipientAvatarPath: string;
    content: string;
    created: Date;
    read?: Date;
   
}

File: sourcecode/aspnet-core-3-api/Entities/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApi.Entities
{
    public class Message
    {
        public int Id { get; set; }

        public int SenderId { get; set; }

        //public int ChatRoomId { get; set; }

        public int RecipientId { get; set; }

        public string Content { get; set; }

        public DateTime Created { get; set; } = DateTime.Now;

        public DateTime? Read { get; set; }

        public bool SenderDeleted { get; set; }

        public bool RecipientDeleted { get; set; }
    }
}
